fabtests fail to compile on MacOS Ventura 13.5.1 (Apple clang version 12.0.0)

**Describe the bug**
After installing libfabric 1.19.0 with `brew install libfabric` (or cloning, compiling and installing libfabric from source) on MaxOs Ventura 13.5.1, I try to compile fabtests but get the following error when I run make

```
$ make
/Library/Developer/CommandLineTools/usr/bin/make  all-am
  CC       functional/av_xfer.o
In file included from functional/av_xfer.c:35:
In file included from ./include/shared.h:59:
In file included from ./include/ft_osd.h:38:
./include/unix/osd.h:71:9: error: implicit declaration of function 'recv' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        return recv(fd, buf, count, flags);
               ^
./include/unix/osd.h:77:9: error: implicit declaration of function 'send' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        return send(fd, buf, count, flags);
               ^
2 errors generated.
make[1]: *** [functional/av_xfer.o] Error 1
```


**To Reproduce**
Steps to reproduce the behavior:
Installing libfabric from source (I tried with `brew install libfabric` as well and the same error occurs) :
```
$ git clone https://github.com/ofiwg/libfabric && cd libfabric
$ git checkout v1.19.0
$ ./autogen.sh
$ ./configure
$ make && make install
```

To build fabtests
```
$ cd fabtests
$ ./autogen.sh
$ ./configure --with-libfabric=/usr/local
$ make
```


**Expected behavior**
fabtests compile correctly.

**Output**
```
$ make
/Library/Developer/CommandLineTools/usr/bin/make  all-am
  CC       functional/av_xfer.o
In file included from functional/av_xfer.c:35:
In file included from ./include/shared.h:59:
In file included from ./include/ft_osd.h:38:
./include/unix/osd.h:71:9: error: implicit declaration of function 'recv' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        return recv(fd, buf, count, flags);
               ^
./include/unix/osd.h:77:9: error: implicit declaration of function 'send' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        return send(fd, buf, count, flags);
               ^
2 errors generated.
make[1]: *** [functional/av_xfer.o] Error 1
```


**Environment:**
MacOS Ventura 13.5.1
Compiling with Apple clang version 12.0.0 (the `gcc` command is aliased to clang on MacOS)
```
$ gcc --version
Configured with: --prefix=/Library/Developer/CommandLineTools/usr --with-gxx-include-dir=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/4.2.1
Apple clang version 12.0.0 (clang-1200.0.31.1)
Target: x86_64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Additional context**
* This does not happen when I compiled on Ubuntu Server 22.04 with gcc 11.4.0 with the same above steps.
* The error goes away if I add a line `#include <sys/socket.h>` in `fabtests/include/unix/osd.h` among the other includes (see below diff)
```
$ git diff
diff --git a/fabtests/include/unix/osd.h b/fabtests/include/unix/osd.h
index 1fd29f619..ec8ca1020 100644
--- a/fabtests/include/unix/osd.h
+++ b/fabtests/include/unix/osd.h
@@ -36,6 +36,7 @@
 #include <complex.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <sys/socket.h>

 #ifndef SOCKET
 #define SOCKET int
```

--- a/fabtests/include/unix/osd.h
+++ b/fabtests/include/unix/osd.h
@@ -36,6 +36,7 @@
 #include <complex.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <sys/socket.h>
 
 #ifndef SOCKET
 #define SOCKET int

